### PR TITLE
Unit Tests [WIP]

### PR DESCRIPTION
**Pull request by MiGraber**
@ptl93 
In the description of makeBasicReportTask. It says that target can be NULL
but when running this code:
```
data("airquality")
basic.report.task = makeBasicReportTask(id = "test.report", data = airquality, target = NULL)
basic.report = makeBasicReport(basic.report.task)
```
The second line throws an error:
See: https://travis-ci.org/ptl93/AEDA/jobs/356591313#L1823-L1830
Everything after is not related to this.

### CODE REVIEW

**Comment by tuanle618** [comment]
@MiGraber I fixed the error and pushed the update.

File: R/getClusterAnalysis.R [hunk 1]
```diff
@@ -142,12 +143,12 @@ getClusterAnalysis = function(data, num.features, method, par.vals, random.seed,
     } else if (method == "cluster.diana") {
       cluster.method = "diana"
     }
-    out.clust = do.call(eclust, args = append(list(x = num.data, FUNcluster = cluster.method, verbose = FALSE, k = 4), par.vals))
```
Comment (MiGraber) [inline R/getClusterAnalysis.R:158]: @ptl93 are you fine with this change: Nmber of clusters are not fixed to 4. (look at the whole file to see all the changes)

Comment (tuanle618) [reply R/getClusterAnalysis.R:158]: Hey, yeah sure. I forgot to change that. Indeed, you are right to let it flexible and not determine the number of groups for the hierarchical clustering algorithms. I am fine with the chane. Thx

Comment (tuanle618) [reply R/getClusterAnalysis.R:158]: I merged my branch and caused a few merge conflicts in getClusterAnalysis. Would be good if you can add your version of number of groups within the hierarchical methods.

**Pull request merged**